### PR TITLE
(PDB-2865) Add quick-retries to PDB command processing

### PR DESCRIPTION
**Pull request by ajroetker**
Current versions of PuppetDB incur a high cost when a command is
retried. The command gets written to a scheduler queue, then re-enqueued
in the regular queue, then reprocessed. A very common case is a database
transaction issue, where two commands are attempting to update the same
data. One thread wins, one loses and has to be retried. We should
implement a quicker "inner loop" retry, where we don't incur the I/O
overhead in this common case. When the quick retries fail, we can then
go through the normal retry logic.

This commit implements this quick retry in command processing (separate
from the queue logic) and removes the exponential backoff for the
traditional retry logic in favor of a set 10 minute retry time.

### CODE REVIEW

File: src/puppetlabs/puppetdb/command.clj [hunk 1]
```diff
@@ -410,9 +410,24 @@
           :producer-timestamp (-> cmd :payload :producer_timestamp)}
          (when ex {:exception ex})))
 
+(defn call-with-quick-retry [num-retries f]
+  (loop [n num-retries]
+    (let [result (try+
```
Comment (senior) [inline src/puppetlabs/puppetdb/command.clj]: I think it makes sense to have a success indicator, but I think we want to return the results of invoking (f) as the return value of call-with-quick-retry

File: test/puppetlabs/puppetdb/command_test.clj [hunk 2]
```diff
@@ -227,22 +221,64 @@
...
+  (with-redefs [metrics.meters/mark! (call-counter)]
+    (let [publish (call-counter)]
       (testing "Exception with stacktrace, no more retries"
         (let [log-output (atom [])]
           (binding [*logger-factory* (atom-logger log-output)]
             (mql/handle-command-retry (make-cmd mql/maximum-allowable-retries) (RuntimeException. "foo") publish))
           (is (= (get-in @log-output [0 1]) :error))
           (is (instance? Exception (get-in @log-output [0 2]))))))))
 
+
```
Comment (senior) [inline test/puppetlabs/puppetdb/command_test.clj]: Extra whitespace

**Comment by senior** [comment]
Minor comments, but looks good!

File: src/puppetlabs/puppetdb/command.clj (hunk 1 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/command.clj]: fixed that up @senior

**Pull request merged**